joern-cli/frontends/php2cpg/src/main/scala/io/joern/php2cpg/astcreation/AstCreator.scala

```diff
@@ -147,16 +146,19 @@ class AstCreator(filename: String, phpAst: PhpFile, global: Global)
     callAst(callNode, args)
   }
 
-  private def thisParamAstForMethod(lineNumber: Option[Integer]): Ast = {
+  private def thisParamAstForMethod(originNode: PhpNode): Ast = {
     val typeFullName = registerType(scope.getEnclosingTypeDeclTypeName.getOrElse(TypeConstants.Any))
-    val thisNode = NewMethodParameterIn()
-      .name(NameConstants.This)
-      .code(NameConstants.This)
-      .evaluationStrategy(EvaluationStrategies.BY_SHARING)
-      .typeFullName(typeFullName)
-      .dynamicTypeHintFullName(typeFullName :: Nil)
-      .lineNumber(lineNumber)
-      .index(0)
+
+    val thisNode = parameterInNode(
+      originNode,
+      name = NameConstants.This,
+      code = NameConstants.This,
+      index = 0,
+      isVariadic = false,
+      evaluationStrategy = EvaluationStrategies.BY_SHARING,
+      typeFullName = typeFullName
+    ).dynamicTypeHintFullName(typeFullName :: Nil)
+    // TODO Add dynamicTypeHintFullName to parameterInNode param list
```

Comment:
I left this as a TODO for this PR since it's already getting big and this change would affect other frontends as well. Was this intentionally omitted? If so, then keeping the code as is and just removing the TODO would be the way to go.

Reply:
I didn't add that parameter because I wanted to keep the PRs with as minimal as possible, but I don't see why not add it as an optional parameter in the near future